vcs/streamline.py

```diff
@@ -0,0 +1,1009 @@
+#               permission of the author.                                     #
+#                                                                             #
+# Authors:      PCMDI Software Team                                           #
+#               Lawrence Livermore NationalLaboratory:                        #
+#               support@pcmdi.llnl.gov                                        #
+#                                                                             #
+# Description:  Python command wrapper for VCS's streamline graphics method.  #
+#                                                                             #
+# Version:      4.0                                                           #
+#                                                                             #
+###############################################################################
+#
+#
+#
+import vcs
+import VCS_validation_functions
+import cdtime
+from xmldocs import scriptdocs
+
+
+def load(nm, json_dict={}):
+    return
+
+
+def process_src(nm, code):
+
+    # Takes VCS script code (string) as input and generates streamline gm from it
+    try:
+        gm = Gs(nm)
+    except:
+        gm = vcs.elements["streamline"][nm]
+    # process attributes with = as assignement
+    for att in ["projection",
+                "xticlabels#1", "xticlabels#2",
+                "xmtics#1", "xmtics#2",
+                "yticlabels#1", "yticlabels#2",
+                "ymtics#1", "ymtics#2",
+                "xaxisconvert", "yaxisconvert",
+                "datawc_tunits",
+                "datawc_tunits",
+                "datawc_calendar",
+                "Tl", "streamline_align", "streamline_type", "ref_streamline",
+                ]:
+        i = code.find(att)
+        if i == -1:
+            continue
+        j = code[i:].find(",") + i
+        if j - i == -1:  # last one no comma
+            j = None
+        scode = code[i:j]
+        sp = scode.split("=")
+        nm = sp[0].strip()
+        nm = nm.replace("#", "")
+        if nm == "streamline_align":
+            nm = "alignement"
+            if sp[1] == "c":
+                sp[1] = "center"
+            elif sp[1] == "h":
+                sp[1] = "head"
+            elif sp[1] == "t":
+                sp[1] = "tail"
+        elif nm == "ref_streamline":
+            nm = "reference"
+        elif nm == "streamline_type":
+            nm = "type"
+        elif nm == "datawc_tunits":
+            nm = "datawc_timeunits"
+        try:
+            # int will be converted
+            setattr(gm, nm, int(sp[1]))
+        except:
+            try:
+                # int and floats will be converted
+                setattr(gm, nm, eval(sp[1]))
+            except:
+                # strings
+                try:
+                    setattr(gm, nm, sp[1])
+                except:
+                    pass  # oh well we stick to default value
+    # Tl
+    i = code.find("Tl")
```

Comment:
@danlipsa you get yank out all the code for reading in streamlines from .scr files since they didn't exist before and we wont generate .scr files for them.